pilot/pkg/credentials/kube/secrets.go

```diff
@@ -218,6 +219,20 @@ func (s *CredentialsController) GetCaCert(name, namespace string) (cert []byte,
 	return extractRoot(k8sSecret)
 }
 
+func (s *CredentialsController) GetDockerCredential(name, namespace string) ([]byte, error) {
+	k8sSecret, err := s.secretLister.Secrets(namespace).Get(name)
+	if err != nil {
+		return nil, fmt.Errorf("secret %v/%v not found", namespace, name)
+	}
+	if k8sSecret.Type != v1.SecretTypeDockerConfigJson {
```

Comment:
Though from the undeylying implement k8sSecret can not be nil, it is better to check